db/src/tx.rs

```diff
@@ -824,9 +813,9 @@ impl TxObserver {
     }
 }
 
-#[derive(Default)]
+#[derive(Default, Clone)]
 pub struct TxObservationService {
-    observers: BTreeMap<String, TxObserver>,
+    observers: BTreeMap<String, Rc<TxObserver>>,
```

Comment:
So, this can be `Arc`, no problem, but being `Weak` presents difficulties. In order for us to hold a `Weak` then something, somewhere needs to hold on to a strong reference of the `TxObserver`. This is fine for Materialized views, but for client observers, the `TxObserver` is created in the FFI boundary as we cannot create an instance of the observer inside the clients (can't create Rust objects in Swift, only create Rust object to wrap Swift objects once they hit Rust). This means that the `Arc` wrapping the `TxObserver` will fall out of scope immediately after registration and upgrading the weak reference on `notify` will fail.

I am therefore going to leave the observation service fully owning the `Arc`.